Stratis.Bitcoin/Features/Wallet/Helpers/SendTransactionComparer.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Stratis.Bitcoin.Features.Wallet.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace Stratis.Bitcoin.Features.Wallet.Helpers
+{
+    /// <summary>
+    ///  Used for comparing a list of TransactionItemModel to each other.
+    /// </summary>
+    /// <remarks>
+    ///  This specific comparer has been made to filter out duplicate payment listings in the case of a transaction with multiple inputs.
+    ///  This causes multiple items inside the wallet data that all have the same amount, transaction/spending id and payment details.
+    ///  However not all fields on this model are equal to each other so we only compare the ones we need.
+    ///  See: https://github.com/stratisproject/Breeze/issues/175
+    /// </remarks>
+    public class SendTransactionComparer : IEqualityComparer<TransactionItemModel>
+    {
+        /// <summary>
+        ///  Determines whether the specified objects are equal.
+        /// </summary>
+        /// <param name="x">The first object of type TransactionItemModel to compare.</param>
+        /// <param name="y">The second object of type TransactionItemModel to compare.</param>
+        /// <returns>true if the specified objects are equal; otherwise, false.</returns>  
+        public bool Equals(TransactionItemModel x, TransactionItemModel y)
+        {
+            if (x == null && x == null)
+                return true;
+            else if (x == null || y == null)
+                return false;
+
+            var propertiesAreEqual = (x.Id == y.Id &&
+                x.ConfirmedInBlock == y.ConfirmedInBlock &&
+                x.Type == y.Type && x.Amount == y.Amount && x.Payments.Count == y.Payments.Count);
+
+            if (!propertiesAreEqual)
+            {
+                return false;
+            }
+
+            foreach (var payment in x.Payments)
+            {
+                if (!y.Payments.Any(w => w.Amount == payment.Amount && w.DestinationAddress == payment.DestinationAddress))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a hash code for the specified object.
+        /// </summary>
+        /// <param name="obj">The System.Object for which a hash code is to be returned.</param>
+        /// <returns>A hash code for the specified object.</returns>
+        public int GetHashCode(TransactionItemModel obj)
+        {
+            if (obj == null)
+                throw new ArgumentNullException(nameof(obj));
+
+            var hashValue = obj.Id.GetHashCode() ^
+                obj.ConfirmedInBlock.GetHashCode() ^
+                obj.Type.GetHashCode() ^
+                obj.Amount.GetHashCode();
+
+            foreach (var payment in obj.Payments)
+            {
+                hashValue = hashValue ^ payment.GetHashCode();
```

Comment:
we could have had a comparer for payments but that's a bit overkill. this is fine